community_tasks/swiss_legal_evals.py

```diff
@@ -0,0 +1,402 @@
+            "large",
+        ], "Model size must be either tiny, base, or large"
+        assert seq_len in [128, 512], "Sequence length must be either 128 or 512"
+
+        self.tokenizer = AutoTokenizer.from_pretrained(
+            f"Elron/bleurt-{model_size}-{seq_len}"
+        )
+        self.model = AutoModelForSequenceClassification.from_pretrained(
+            f"Elron/bleurt-{model_size}-{seq_len}"
+        )
+        self.max_length = seq_len
+        self.model.eval()
+
+    def compute(self, golds: list[str], predictions: list[str], **kwargs) -> float:
+        """Uses the stored BLEURT scorer to compute the score on the current sample.
+
+        Args:
+            golds (list[str]): Reference targets
+            predictions (list[str]): Predicted strings
+
+        Returns:
+            float: Score over the current sample's items.
+        """
+        if len(predictions) == 1:
+            predictions = predictions * len(golds)
+        inputs = self.tokenizer(
+            golds,
+            predictions,
+            return_tensors="pt",
+            padding=True,
+            truncation=True,
+            max_length=self.max_length,
+        )
+        if any(len(encoding) == self.max_length for encoding in inputs["input_ids"]):
+            hlog_warn(
+                f"Some inputs were truncated to max_length={self.max_length} in BLEURT scoring"
+            )
+        scores = self.model(**inputs)[0].squeeze()
+        return scores.item()
+
+
+def bleurt(model_size: str = "tiny", seq_len: int = 512):
+    return SampleLevelMetric(
+        metric_name="bleurt",
+        sample_level_fn=BLEURT(model_size=model_size, seq_len=seq_len).compute,
+        category=MetricCategory.GENERATIVE,
+        use_case=MetricUseCase.TRANSLATION,
+        corpus_level_fn=np.mean,
+        higher_is_better=True,
+    )
+
+
+# EVALS WITH SUBSET
+# This is how you create a subset task (like MMLU), which has several subset
+# each being its own evaluation task.
+
+
+def create_translation_pairs(langs_list: list) -> list[tuple]:
+    """
+    Create all possible translation pairs from a given list of languages.
+
+    Args:
+    langs_list (list): A list of languages.
+
+    Returns:
+    lang_pairs_list (list): A list of tuples representing a translation pair.
+    """
+    lang_pairs_list = []
+    for i, lang1 in enumerate(langs_list):
+        for lang2 in langs_list[i + 1 :]:
+            lang_pairs_list.append((lang1, lang2))
+            lang_pairs_list.append((lang2, lang1))
+    return lang_pairs_list
+
+
+@dataclass
+class LevelConfig:
+    name: str
+    text_col_name: str
+    prompt_prefix: str
+    metadata_cols: list[str]
+
+
+@dataclass
+class DatasetConfig:
+    name: str
+    hf_repo: str
+    languages: list[str]
+    subsets: dict[str, LevelConfig]
+
+    def __post_init__(self):
+        self.translation_pairs = create_translation_pairs(self.languages)
+
+
+SwissDecisionSummaryTranslations = DatasetConfig(
```

Comment:
Would be worth adding a very small desc of all datasets if possible

Reply:
Done